library/psa_crypto.c

```diff
@@ -3131,7 +3156,761 @@ psa_status_t psa_asymmetric_decrypt( mbedtls_svc_key_id_t key,
+                                                    MBEDTLS_PSA_RANDOM_STATE,
+                                                    &operation->restart_ctx ) );
+    }
+
+    if( status == PSA_SUCCESS )
+    {
+        status =  mbedtls_to_psa_error(
+                        mbedtls_mpi_write_binary( &r,
+                                                  signature,
+                                                  operation->coordinate_bytes )
+                );
+
+        if( status != PSA_SUCCESS )
+            goto exit;
+
+        status =  mbedtls_to_psa_error(
+                        mbedtls_mpi_write_binary( &s,
+                                                  signature +
+                                                  operation->coordinate_bytes,
+                                                  operation->coordinate_bytes )
+                );
+
+        if( status != PSA_SUCCESS )
+            goto exit;
+
+        *signature_length = operation->coordinate_bytes * 2;
+
+        status = PSA_SUCCESS;
+    }
+
+exit:
+
+    mbedtls_mpi_free(&r);
+    mbedtls_mpi_free(&s);
+    return status;
+
+ #else
+
+    ( void ) operation;
+    ( void ) signature;
+    ( void ) signature_size;
+    ( void ) signature_length;
+
+    return( PSA_ERROR_NOT_SUPPORTED );
+
+#endif /* defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) ||
+        * defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA) &&
+        * defined( MBEDTLS_ECP_RESTARTABLE ) */
+}
+
+psa_status_t mbedtls_psa_sign_hash_abort(
+                    mbedtls_psa_sign_hash_interruptible_operation_t *operation )
+{
+
+#if (defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) || \
+     defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA) ) && \
+     defined( MBEDTLS_ECP_RESTARTABLE )
+
+    if( operation->ctx )
+    {
+        mbedtls_ecdsa_free( operation->ctx );
+        mbedtls_free( operation->ctx );
+        operation->ctx = NULL;
+    }
+
+    mbedtls_ecdsa_restart_free( &operation->restart_ctx );
+
+    operation->num_ops = 0;
+
+    return PSA_SUCCESS;
+
+#else
+
+    ( void ) operation;
+
+    return( PSA_ERROR_NOT_SUPPORTED );
+
+#endif /* defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) ||
+    * defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA) &&
+    * defined( MBEDTLS_ECP_RESTARTABLE ) */
+}
+
+psa_status_t mbedtls_psa_verify_hash_start(
+                mbedtls_psa_verify_hash_interruptible_operation_t *operation,
+                const psa_key_attributes_t *attributes,
+                const uint8_t *key_buffer, size_t key_buffer_size,
+                psa_algorithm_t alg,
+                const uint8_t *hash, size_t hash_length,
+                const uint8_t *signature, size_t signature_length )
+{
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+    size_t coordinate_bytes = 0;
+    size_t required_hash_length = 0;
+
+    if( !PSA_KEY_TYPE_IS_ECC( attributes->core.type ) )
+        return PSA_ERROR_NOT_SUPPORTED;
```

Comment:
[non-blocker] This is not tested according to lcov, but it seems testable.

Reply:
#7113